From bcca489771f5c91286d661eb5c26e586a15f7cdb Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Fri, 18 Mar 2022 18:51:28 +0100
Subject: [PATCH 3/3] Enable choice of library version for shell

Default behavior is unchanged, but enables to choose the dynamic version by
passing "libmujs=libmujs.so" to make.

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 Makefile | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 2d6fb959b6b2..c026325fdb7b 100644
--- a/Makefile
+++ b/Makefile
@@ -90,13 +90,15 @@ $(OUT)/libmujs.$(SO_EXT): one.c $(HDRS)
 	@ mkdir -p $(@D)
 	$(CC) $(CFLAGS) $(CPPFLAGS) -fPIC -shared $(LDFLAGS) -o $@ $< -lm
 
-$(OUT)/mujs: $(OUT)/libmujs.o $(OUT)/main.o
+libmujs ?= libmujs.a
+
+$(OUT)/mujs: $(OUT)/main.o $(OUT)/$(libmujs)
 	@ mkdir -p $(@D)
-	$(CC) $(LDFLAGS) -o $@ $^ $(LIBREADLINE) -lm
+	$(CC) $(LDFLAGS) -o $@ $< -L$(OUT) -l:$(libmujs) $(LIBREADLINE) -lm
 
-$(OUT)/mujs-pp: $(OUT)/libmujs.o $(OUT)/pp.o
+$(OUT)/mujs-pp: $(OUT)/pp.o $(OUT)/$(libmujs)
 	@ mkdir -p $(@D)
-	$(CC) $(LDFLAGS) -o $@ $^ -lm
+	$(CC) $(LDFLAGS) -o $@ $< -L$(OUT) -l:$(libmujs) -lm
 
 .PHONY: $(OUT)/mujs.pc
 $(OUT)/mujs.pc:
-- 
2.35.1

